### PR TITLE
FE Delete & Rename shortcut keys

### LINKED ISSUE #258: Fileexplorer: add delete and rename files with "delete" and F2 keys

### PR DESCRIPTION
**Pull request by joeizang**
fixes #258

### CODE REVIEW

File: libs/remix-ui/workspace/src/lib/components/file-explorer.tsx [hunk 1]
```diff
@@ -97,6 +101,98 @@ export const FileExplorer = (props: FileExplorerProps) => {
     }
   }, [treeRef.current])
 
+  useEffect(() => {
+    const performDeleteion = async () => {
```
Comment (Aniket-Engg) [inline libs/remix-ui/workspace/src/lib/components/file-explorer.tsx:105]: typo in `Deleteion`

File: libs/remix-ui/workspace/src/lib/components/file-explorer.tsx [hunk 2]
```diff
@@ -97,6 +101,98 @@ export const FileExplorer = (props: FileExplorerProps) => {
...
+          if (eve.key === 'Backspace') {
+            setState((prevState) => {
+              return { ...prevState, deleteKey: true }
+            })
+            performDeleteion()
+            return
+          }
+        }
+      }
+      const deleteKeyPressUpHandler = async (eve: KeyboardEvent) => {
```
Comment (Aniket-Engg) [inline libs/remix-ui/workspace/src/lib/components/file-explorer.tsx:134]: Why do you need to add listeners for both `keyup` & `keydown`?

File: libs/remix-ui/workspace/src/lib/components/file-explorer.tsx [hunk 3]
```diff
@@ -97,6 +101,98 @@ export const FileExplorer = (props: FileExplorerProps) => {
...
+        if (eve.key === 'Delete' ) {
+          setState((prevState) => {
+            return { ...prevState, deleteKey: false }
+          })
+          return
```
Comment (Aniket-Engg) [inline libs/remix-ui/workspace/src/lib/components/file-explorer.tsx:139]: should `performDeleteion()` be called here too?

File: libs/remix-ui/workspace/src/lib/components/file-explorer.tsx [hunk 4]
```diff
@@ -357,6 +453,7 @@ export const FileExplorer = (props: FileExplorerProps) => {
   }
 
   const handleTreeClick = (event: SyntheticEvent) => {
+    console.log({ props, state })
```
Comment (Aniket-Engg) [inline libs/remix-ui/workspace/src/lib/components/file-explorer.tsx:456]: remove log

**Review by Aniket-Engg — COMMENTED** [review]
Funtionality wise it looks fine

File: libs/remix-ui/workspace/src/lib/components/file-explorer.tsx (hunk 1 above)
Comment (joeizang) [reply libs/remix-ui/workspace/src/lib/components/file-explorer.tsx:105]: ok

File: libs/remix-ui/workspace/src/lib/components/file-explorer.tsx (hunk 2 above)
Comment (joeizang) [reply libs/remix-ui/workspace/src/lib/components/file-explorer.tsx:134]: Yes, the keyUp updates the state variable for deleteKey and F2Key

File: libs/remix-ui/workspace/src/lib/components/file-explorer.tsx (hunk 3 above)
Comment (joeizang) [reply libs/remix-ui/workspace/src/lib/components/file-explorer.tsx:139]: Yes, its handling the case for either the Delete key or the combination of meta + Backspace

File: libs/remix-ui/workspace/src/lib/components/file-explorer.tsx [hunk 5]
```diff
@@ -97,6 +101,100 @@ export const FileExplorer = (props: FileExplorerProps) => {
...
+          path.push(one.key)
+        })
+        await deletePath(path)
+      }
+    }
+
+    if (treeRef.current) {
+      const deleteKeyPressHandler = async (eve: KeyboardEvent) => {
+        if (eve.key === 'Delete' ) {
+          (window as any)._paq.push(['trackEvent', 'fileExplorer', 'deleteKey'])
```
Comment (Aniket-Engg) [inline libs/remix-ui/workspace/src/lib/components/file-explorer.tsx:118]: it will be better to create a variable for `(window as any)`

Comment (Aniket-Engg) [reply libs/remix-ui/workspace/src/lib/components/file-explorer.tsx:118]: also `_paq.push` takes 4 args

Comment (joeizang) [reply libs/remix-ui/workspace/src/lib/components/file-explorer.tsx:118]: That's right.

**Pull request merged**